### ISSUE TITLE
BigQuery: Normalization modes not interpreted as keywords in `NORMALIZE` function

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3085]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Normalization modes to BigQuery's `NORMALIZE` and `NORMALIZE_AND_CASEFOLD ` are not correctly understood by SQLFluff

https://cloud.google.com/bigquery/docs/reference/standard-sql/string_functions#normalize

### Expected Behaviour

SQLFluff should understand Normalization modes as keywords.

### Observed Behaviour

Normalization modes are understood as identifiers and SQLFluff tries to apply identifier rules.

### How to reproduce

```sql
SELECT
    col1,
    NORMALIZE('\u00ea', NFD) AS a,
    NORMALIZE('\u0065\u0302', NFD) AS b,
    NORMALIZE_AND_CASEFOLD('\u00ea', NFD) AS c,
    NORMALIZE_AND_CASEFOLD('\u0065\u0302', NFD) AS d;
```

https://online.sqlfluff.com/fluffed?sql=H4sIANzlVmIC_wt29XF1DuHlUgCC5PwcQx0I088_yNfRxzPKVUM9ptTAIDVRXUfBz81FU8ExWCERuxozUyBpbGCEpDIJXWW8o59LvLNjsKubv48LFqOTidCAxZ4Ua6AeiE4YCQBrP0pS2AAAAA%3D%3D&dialect=bigquery

### Dialect

bigquery

### Version

0.12 -> webversion

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
